File: java/108.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        if(nums.length == 0){
            return null;
        }
        int l = nums.length;
        int lo = 0;
        int hi = l-1;
        // int mid = l/2;
        
        return toBST(nums,lo,hi);
    }
    public TreeNode toBST(int[] nums,int lo,int hi){
        if(lo > hi){
            return null;
        }
        if(lo == hi){
            return new TreeNode(nums[lo]);
        }
        int mid = (lo+hi)/2;
        TreeNode root = new TreeNode(nums[mid]);
        root.left = toBST(nums,lo,mid-1);
        root.right = toBST(nums,mid+1,hi);
        return root;
    }
}

File: go/108.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sortedArrayToBST(nums []int) *TreeNode {
	if len(nums) == 0 {
		return nil
	}
	l := len(nums) / 2
	root := &TreeNode{nums[l], nil, nil}
	root.Left = sortedArrayToBST(nums[:l])
	root.Right = sortedArrayToBST(nums[l+1:])
	return root
}